### PR TITLE
Switch the UI to use Flutter Web

### LINKED ISSUE #25: Prototype Flutter Web UI Skeleton
We should prototype using [Flutter](https://flutter.dev/) instead of raw HTML, CSS, and Javascript for our UI. Particularly we can use [Flutter Web](https://flutter.dev/web) to compile a cross-platform, mobile-first application into a single Javascript file.

To successfully set this up we will need:
- A Java servlet filter to redirect non-static content (or API requests) to a single HTML file that pulls in the single Javascript compiled application file. This is partly required because Flutter Web (like many modern web frameworks) performs in-browser routing. What this means is that for all request paths we should serve the application (HTML + Javascript file) which will then examine the browser route to determine which application pages to render.
- A way of automating the compilation of the Flutter Web app, copying compiled resources into the webapp directory, and then packaging them into the war when `mvn package` is run.

Please note this is purely exploratory until we agree on a UI framework to use.

### PR DESCRIPTION
**Pull request by KoopaKing**
closes: #25

### CODE REVIEW

File: frontend/ui/lib/services/in_memory_trip_service.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import 'dart:async';
+import 'dart:core';
+import 'package:tripmeout/model/trip.dart';
+import 'package:tripmeout/services/trip_service.dart';
+
+class InMemoryTripService implements TripService {
+  final Map<String, Trip> tripsById = Map();
+
+  Future<List<Trip>> listTrips() async {
```
Comment (cli1903) [inline frontend/ui/lib/services/in_memory_trip_service.dart:14]: do these functions not need override annotations?

Comment (KoopaKing) [reply frontend/ui/lib/services/in_memory_trip_service.dart:14]: Good call.

**Review by anfee1 — APPROVED** [review]
I've looked over it twice and I think it LGTM.

**Review by cli1903 — APPROVED** [review]
looks good as far as I can tell

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+SHELL=/bin/bash
+FLUTTER_CLI?=flutter
+MAVEN_CLI?=mvn
+FLUTTER_APP_ROOT?=frontend/ui
+MAVEN_FRONTEND_ROOT?=frontend
+CLANG_FORMAT?=clang-format
+
+# Is there a better way to do this??
+# From https://stackoverflow.com/questions/4036191
+rwildcard=$(wildcard $1$2) $(foreach d,$(wildcard $1*),$(call rwildcard,$d/,$2))
```
Comment (bsidhom) [inline Makefile:11]: That's unfortunate. I could have sworn that `make` supported `**/*` globs, but I wasn't able to get it working.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+format:
+	$(CLANG_FORMAT) --style=Google -i --sort-includes $(call rwildcard,frontend/,*.java)
+	$(FLUTTER_CLI) format $(FLUTTER_APP_ROOT)
```
Comment (bsidhom) [inline Makefile:15]: FYI, this gives me a warning to use `dart format` instead.

**Pull request merged**